src/main/java/com/nike/cerberus/domain/environment/EnvironmentData.java

```diff
@@ -0,0 +1,160 @@
+        return adminIamRoleArn;
+    }
+
+    public void setAdminIamRoleArn(String adminIamRoleArn) {
+        this.adminIamRoleArn = adminIamRoleArn;
+    }
+
+    public String getRootIamRoleArn() {
+        return rootIamRoleArn;
+    }
+
+    public void setRootIamRoleArn(String rootIamRoleArn) {
+        this.rootIamRoleArn = rootIamRoleArn;
+    }
+
+    public String getCmsIamRoleArn() {
+        return cmsIamRoleArn;
+    }
+
+    public void setCmsIamRoleArn(String cmsIamRoleArn) {
+        this.cmsIamRoleArn = cmsIamRoleArn;
+    }
+
+    public String getDatabasePassword() {
+        return databasePassword;
+    }
+
+    public void setDatabasePassword(String databasePassword) {
+        this.databasePassword = databasePassword;
+    }
+
+    public LinkedList<CertificateInformation> getCertificateData() {
+        return certificateInfoList;
+    }
+
+    public void addNewCertificateData(CertificateInformation certificateData) {
+        getCertificateData().add(certificateData);
+    }
+
+    public void removeCertificateInformationByName(String identityManagementCertificateName) {
+        List<CertificateInformation> certsInfoToBeDeleted = certificateInfoList.stream()
+                .filter(certificateInformation ->
+                        certificateInformation.getCertificateName().equals(identityManagementCertificateName))
+                .collect(Collectors.toList());
+
+        certificateInfoList.removeAll(certsInfoToBeDeleted);
+    }
+
+    public Map<Regions, RegionData> getRegionData() {
+        return regionData;
+    }
+
+    public void addRegionData(Regions region, RegionData rData) {
+        regionData.put(region, rData);
+    }
+
+    /**
+     * @return List of all the regions that are configured for storing config.
+     */
+    @JsonIgnore
+    public List<Regions> getConfigRegions() {
+        List<Regions> configRegions = new LinkedList<>();
+        regionData.forEach((region, rData) -> {
+            if (rData.getEnvironmentDataSecureDataKmsCmkArn().isPresent()
+                    && rData.getConfigBucket().isPresent()) {
+                configRegions.add(region);
+            }
+        });
+        return configRegions;
+    }
+
+    @JsonIgnore
+    public Regions getPrimaryRegion() {
+        final Regions[] primaryRegion = {null};
+        regionData.forEach((region, regionData) -> {
+            if (regionData.isPrimary()) {
+                primaryRegion[0] = region;
+            }
+        });
+        if (primaryRegion[0] == null) {
+        throw new RuntimeException("Failed to find primary region in region specific state");
+    }
+        return primaryRegion[0];
+    }
+
+    public void addCmksCmkForRegion(Regions region, String arn) {
+        RegionData rData;
+        if (!regionData.containsKey(region)) {
+            rData = new RegionData();
+            regionData.put(region, rData);
+        } else {
+            rData = regionData.get(region);
+        }
```

Comment:
Above "getOrCreate" logic could move into the RegionData class?  It seems like a better owner for that functionality.

Reply:
- [ ] move functionality to RegionData

Reply:
region data here is a map

Reply:
I think @tlisonbee means the `rData` variable which is an instance of the RegionData class.

Reply:
Again, unless I am taking crazy pills there is no logic that is in this method that would make sense to move into the RegionData class. please provide examples if you disagree.

Reply:
I guess it doesn't even matter I deleted this method as its not used anymore